services/brig/test/integration/API/Provider.hs

```diff
@@ -169,6 +170,58 @@ testRegisterProvider db brig = do
         assertEqual "description" defProviderDescr (providerDescr p)
         assertEqual "profile" (ProviderProfile p) pp
 
+-- | Step-by-step registration procedure with verification
+-- of pre- and post-conditions, using an internal endpoint
+-- instead of accessing the DB directly
+testRegisterProviderInternal :: Brig -> Http ()
+testRegisterProviderInternal brig = do
+    email <- mkSimulatorEmail "success"
+
+    let new = defNewProvider email
+
+    _rs <- registerProvider brig new <!!
+        const 201 === statusCode
+
+    let Just npr = decodeBody _rs :: Maybe NewProviderResponse
+    -- Since a password was given, none should have been generated
+    liftIO $ assertBool "password" (isNothing (rsNewProviderPassword npr))
+    let pid = rsNewProviderId npr
+
+    -- No login possible directly after registration
+    loginProvider brig email defProviderPassword !!!
+        const 403 === statusCode
+
+    -- Activate email
+    _rs <- getProviderActivationCodeInternal brig email <!!
+        const 200 === statusCode
+
+    let Just pair = decodeBody _rs :: Maybe Code.KeyValuePair
+    activateProvider brig (Code.kcKey pair) (Code.kcCode pair) !!!
+        const 200 === statusCode
+
+    -- Login succeeds after activation (due to auto-approval)
+    loginProvider brig email defProviderPassword !!!
+        const 200 === statusCode
+
+    -- Email address is now taken
+    registerProvider brig new !!!
+        const 409 === statusCode
+
+    -- Retrieve full account and public profile
+    -- (these are identical for now).
+    uid <- randomId
+    _rs <- getProvider brig pid <!! const 200 === statusCode
+    let Just p  = decodeBody _rs
+    _rs <- getProviderProfile brig pid uid <!! const 200 === statusCode
+    let Just pp = decodeBody _rs
+    liftIO $ do
```

Comment:
Can you add a comment to `Provider` saying that there are tests that compare fields one-by-one and therefore they have to be updated when there are fields added to `Provider`?